### PR TITLE
feat(HACBS-1763): add GitHub workflow to automate the release

### PR DESCRIPTION
**Pull request by davidmogar**
This commit adds a workflow to populate the repository used to deploy resources to app-interface.

Signed-off-by: David Moreno García <damoreno@redhat.com>

### CODE REVIEW

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -70,13 +61,19 @@ spec:
         - /manager
         args:
         - --leader-elect
+        - --remote-cluster-config-file
```
Comment (scoheb) [inline config/manager/manager.yaml:64]: Will this get overwritten the next time we push?

Comment (davidmogar) [reply config/manager/manager.yaml]: I removed it. Now it's generated in the workflow.

File: .github/workflows/app_interface.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        uses: chrisdickinson/setup-yq@latest
+        with:
+          yq-version: ${{ env.YQ_VERSION }}
+      - name: Update namespaces
+        run: |
+          find . -type f -exec sed -i 's/namespace: system/namespace: ${{ env.NAMESPACE }}/' {} \;
+        working-directory: app-interface-deployments/internal-services
+      - name: Update image
+        run: |
+          sed -i 's/image: controller:latest/image: quay.io\/redhat-appstudio\/internal-services:${{ github.event.pull_request.head.sha }}/' manager.yaml
```
Comment (scoheb) [inline .github/workflows/app_interface.yaml:50]: is this guaranteed to exist at this time?

File: .github/workflows/app_interface.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+          yq -i '.spec.template.spec.containers[0].args += "--remote-cluster-config-file"' manager.yaml
+          yq -i ".spec.template.spec.containers[0].args += \"${{ env.REMOTE_CLIENT_CONFIG_MOUNT_PATH }}/kubeconfig\"" manager.yaml
+          yq -i ".spec.template.spec.containers[0].volumeMounts = [{\"mountPath\": \"${{ env.REMOTE_CLIENT_CONFIG_MOUNT_PATH }}\", \"name\": \"${{ env.REMOTE_CLIENT_CONFIG_SECRET }}\", \"readOnly\": true }]" manager.yaml
+        working-directory: app-interface-deployments/internal-services/manager
+      - name: Push to app-interface-deployments
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git add .
+          git diff-index --quiet HEAD || git commit -m "Automatic push from radhat-appstudio/internal-services"
```
Comment (scoheb) [inline .github/workflows/app_interface.yaml:72]:
```suggestion
          git diff-index --quiet HEAD || git commit -m "Automatic push from redhat-appstudio/internal-services"
```

File: config/rbac/sample.yaml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+b:
```
Comment (scoheb) [inline config/rbac/sample.yaml:1]: still needed?

**Review by scoheb — CHANGES_REQUESTED** [review]

File: .github/workflows/app_interface.yaml (hunk 2 above)
Comment (davidmogar) [reply .github/workflows/app_interface.yaml:50]: It should as the workflow runs after the commit.

File: config/rbac/sample.yaml (hunk 4 above)
Comment (davidmogar) [reply config/rbac/sample.yaml:1]: I didn't even realize. Thanks for catching this up.

**Review by scoheb — APPROVED** [review]

File: .github/workflows/app_interface.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      - name: Copy files to app-interface-deployments
+        run: |
+          mkdir -p internal-services
+          mkdir -p internal-services/crds
+          mkdir -p internal-services/manager
+          mkdir -p internal-services/rbac
+          cp ../internal-services/config/crd/bases/*.yaml internal-services/crds
+          cp ../internal-services/config/manager/manager.yaml internal-services/manager
+          cp ../internal-services/config/rbac/*.yaml internal-services/rbac
+          rm -rf internal-services/rbac/kustomization.yaml
```
Comment (theflockers) [inline .github/workflows/app_interface.yaml:38]: rm -f is enough. No directory being removed.

**Review by theflockers — COMMENTED** [review]
just a few comments to consider.

**Review by happybhati — APPROVED** [review]

File: .github/workflows/app_interface.yaml (hunk 5 above)
Comment (davidmogar) [reply .github/workflows/app_interface.yaml]: Done.

File: .github/workflows/app_interface.yaml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+      - main
+jobs:
+  generate:
+    name: Push internal-services resources
+    runs-on: ubuntu-20.04
+    env:
+      NAMESPACE: stonesoup-int-srvc
+      REMOTE_CLIENT_CONFIG_SECRET: remote-client-config
+      REMOTE_CLIENT_CONFIG_MOUNT_PATH: "/mnt/internal-services/remote-client-config"
+      USERS: "damoreno shebert"
```
[the remark this answers is not in the document]
Comment (davidmogar) [reply .github/workflows/app_interface.yaml]: Added the existing users.

**Pull request merged**